drivers/display/display_rm67162.c

```diff
@@ -0,0 +1,599 @@
+		/* Setup TE pin */
+		ret = gpio_pin_configure_dt(&config->te_gpio, GPIO_INPUT);
+		if (ret < 0) {
+			LOG_ERR("Could not configure TE GPIO (%d)", ret);
+			return ret;
+		}
+
+		ret = gpio_pin_interrupt_configure_dt(&config->te_gpio,
+						      GPIO_INT_EDGE_TO_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure TE interrupt (%d)", ret);
+			return ret;
+		}
+
+		/* Init and install GPIO callback */
+		gpio_init_callback(&data->te_gpio_cb, rm67162_te_isr_handler,
+				BIT(config->te_gpio.pin));
+		gpio_add_callback(config->te_gpio.port, &data->te_gpio_cb);
+
+		/* Setup te pin semaphore */
+		k_sem_init(&data->te_sem, 0, 1);
+	}
+
+	/* Now, enable display */
+	return mipi_dsi_dcs_write(config->mipi_dsi, config->channel,
+				MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+/* Helper to write data to rm67162 via MIPI interface. */
+static int rm67162_write_buf(const struct device *dev, bool first_write,
+			const uint8_t *src, uint32_t len)
+{
+	const struct rm67162_config *config = dev->config;
+	struct rm67162_data *data = dev->data;
+	int ret = 0;
+	uint32_t max_write, wlen;
+	uint8_t cmd;
+
+	/*
+	 * Max write len: one byte is reserved for DSC command, and
+	 * pixels should not be split across transfers
+	 */
+	max_write = ((DSI_TX_MAX_PAYLOAD_BYTE - 1) / data->bytes_per_pixel) *
+						data->bytes_per_pixel;
+	if (first_write) {
+		cmd = MIPI_DCS_WRITE_MEMORY_START;
+	} else {
+		cmd = MIPI_DCS_WRITE_MEMORY_CONTINUE;
+	}
+	while (len > 0) {
+		/* Cap each tx to max DSI APB transfer size */
+		wlen = MIN(max_write, len);
+		ret = mipi_dsi_dcs_write(config->mipi_dsi, config->channel,
+					cmd, src, wlen);
+		if (ret < 0) {
+			return ret;
+		}
+		/* Advance source pointer and decrement remaining */
+		src += wlen;
+		len -= wlen;
+		/* All future commands should use WRITE_MEMORY_CONTINUE */
+		cmd = MIPI_DCS_WRITE_MEMORY_CONTINUE;
+	}
+	return ret;
+}
+
+static int rm67162_write(const struct device *dev, const uint16_t x,
+			 const uint16_t y,
+			 const struct display_buffer_descriptor *desc,
+			 const void *buf)
+{
+	const struct rm67162_config *config = dev->config;
+	struct rm67162_data *data = dev->data;
+	int ret;
+	uint16_t start, end, h_idx;
+	const uint8_t *src;
+	bool first_cmd;
+	uint8_t param[4];
+
+	LOG_DBG("W=%d, H=%d @%d,%d", desc->width, desc->height, x, y);
+
+	/*
+	 * RM67162 runs in MIPI DBI mode. This means we can use command mode
+	 * to write to the video memory buffer on the RM67162 control IC,
+	 * and the IC will update the display automatically.
+	 */
+
+	/* Set column address of target area */
+	/* First two bytes are starting X coordinate */
+	start = x;
+	end = x + desc->width - 1;
+	param[0] = (uint8_t)((start >> 8U) & 0xFFU);
+	param[1] = (uint8_t)((start) & 0xFFU);
+	/* Second two bytes are ending X coordinate */
+	param[2] = (uint8_t)((end >> 8U) & 0xFFU);
+	param[3] = (uint8_t)((end) & 0xFFU);
+	ret = mipi_dsi_dcs_write(config->mipi_dsi, config->channel,
+				MIPI_DCS_SET_COLUMN_ADDRESS, param, 4);
```

Comment:
I would use `sizeof(param)` instead of repeating `4` again. Less likely to be copypasted wrong.